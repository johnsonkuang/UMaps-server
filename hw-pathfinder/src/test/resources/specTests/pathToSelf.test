# tests if graph can correctly identify a path from the node to itself
CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D

AddEdge simpleGraph A B 10.0
AddEdge simpleGraph A C 1.0
AddEdge simpleGraph C D 1.0
AddEdge simpleGraph D B 1.0

ListNodes simpleGraph

ListChildren simpleGraph A

FindPath simpleGraph A A