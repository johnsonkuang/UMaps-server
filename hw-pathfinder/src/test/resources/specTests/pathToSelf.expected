# tests if graph can correctly identify a path from the node to itself
created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph

added edge 10.000 from A to B in simpleGraph
added edge 1.000 from A to C in simpleGraph
added edge 1.000 from C to D in simpleGraph
added edge 1.000 from D to B in simpleGraph

simpleGraph contains: A B C D

the children of A in simpleGraph are: B(10.000) C(1.000)

path from A to A:
total cost: 0.000